src/RouteCollector.php

```diff
@@ -146,36 +160,28 @@ public function getRoutes() : array
      *
      * @throws Exception\DuplicateRouteException on duplicate route detection.
      */
-    private function checkForDuplicateRoute(string $path, array $methods = null) : void
+    private function checkForDuplicateRoute(Route $route) : void
     {
-        if (null === $methods) {
-            $methods = Route::HTTP_METHOD_ANY;
+        if (array_key_exists($route->getName(), $this->routesNames)) {
+            $this->duplicateRouteDetected($this->routesNames[ $route->getName() ]);
+        }
+
+        if (array_key_exists($route->asUniqueRouteIdentity(), $this->routesIdentities)) {
```

Comment:
I am not sure if it is sufficient.

Let consider situation:
```
$app->any('/path', ...);
$app->get('/path, ...);
```

for the first we will have id: `/path^ANY` and for the second `/path^GET` and here we are not gonna get any issue, but before, I believe, we got the exception.

The same if we have:
```
$app->route('/path', ..., ['GET', 'POST']);
$app->post('/path', ...);
```

etc.

Unless I am missing something, but it looks like you always expect the same path and methods, and only in that case you throw exception.

Reply:
Yep, was not covered by tests. Added 2 more tests to cover this case.